### PR TITLE
feat(lyrics-plus): playbar button

### LINKED ISSUE #2314: Add an option to access Lyrics Plus from default lyrics button
### 📝 Provide a description of the new feature

I'm a first time lyrics plus user and i'm blown a way by it, i just thought it would probably nice to have an option where you can access Lyrics Plus from the default lyrics button at the bottom instead of adding a new tab at the sidebar or we can just hide the tab with sidebar config. Thank you!

### ➕ Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ohitstom**
resolves #2314.

image
image

### CODE REVIEW

**Comment by ohitstom** [comment]
> use event listener on localstorage to detect changes and run functions based on value

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	if (!Spicetify.Platform) {
+		setTimeout(PlaybarButton, 300);
+		return;
+	}
+
+	const button = new Spicetify.Playbar.Button(
+		"Lyrics Plus",
+		`<svg role="img" height="16" width="16" aria-hidden="true" viewBox="0 0 16 16" data-encore-id="icon" class="Svg-sc-ytk21e-0 Svg-img-16-icon">
+	  <path d="M13.426 2.574a2.831 2.831 0 0 0-4.797 1.55l3.247 3.247a2.831 2.831 0 0 0 1.55-4.797zM10.5 8.118l-2.619-2.62A63303.13 63303.13 0 0 0 4.74 9.075L2.065 12.12a1.287 1.287 0 0 0 1.816 1.816l3.06-2.688 3.56-3.129zM7.12 4.094a4.331 4.331 0 1 1 4.786 4.786l-3.974 3.493-3.06 2.689a2.787 2.787 0 0 1-3.933-3.933l2.676-3.045 3.505-3.99z"></path>
+	</svg>`,
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:11]:
```suggestion
	const button = new Spicetify.Playbar.Button(
		"Lyrics Plus",
		"lyrics",
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		() => {
+			if (Spicetify.Platform.History.location.pathname !== "/lyrics-plus") {
+				Spicetify.Platform.History.push("/lyrics-plus");
+			} else {
+				Spicetify.Platform.History.goBack();
+			}
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:17]:
```suggestion
			Spicetify.Platform.History.location.pathname !== "/lyrics-plus" ? Spicetify.Platform.History.push("/lyrics-plus") : Spicetify.Platform.History.goBack()
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		}
+	);
+
+	const initialValue = Spicetify.LocalStorage.get("lyrics-plus:visual:playbar-button");
+	if (initialValue === "true") {
+		setPlaybarButton();
+	}
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:24]:
```suggestion
	if (initialValue === "true") setPlaybarButton();
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	window.addEventListener("lyrics-plus", event => {
+		if (event.detail.name === "playbar-button") {
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:27]:
```suggestion
		if (event.detail?.name === "playbar-button") {
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+				  display: none !important; 
+			  }
+			  li[data-id="/lyrics-plus"] {
+				  display: none;
+			  }
+		  `;
+
+		const style = document.createElement("style");
+		style.innerHTML = css;
+		style.classList.add("lyrics-plus:visual:playbar-button");
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:44]: Create the stylesheet outside the scope of this function. Then on each of the create and remove button function you'll only need to run `document.head.appendChild(style);` and `style.remove()` subsequently,

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		document.head.appendChild(style);
+		button.register();
+	}
+
+	function removePlaybarButton() {
+		const style = document.head.querySelector(".lyrics-plus\\:visual\\:playbar-button");
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:50]: Ditto.

**Review by kyrie25 — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/PlaybarButton.js (hunk 1 above)
Comment (ohitstom) [reply CustomApps/lyrics-plus/PlaybarButton.js:11]: i disagree
image

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	window.addEventListener("lyrics-plus", event => {
+		if (event.detail?.name === "playbar-button") {
+			event.detail.value ? setPlaybarButton() : removePlaybarButton();
+		}
+	});
+
+	if (Spicetify.Platform.History.location.pathname === "/lyrics-plus") button.active = true;
+	Spicetify.Platform.History.listen(location => {
+		location.pathname === "/lyrics-plus" ? (button.active = true) : (button.active = false);
+	});
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:42]:
```suggestion
	Spicetify.Platform.History.listen(location => button.active = location.pathname === "/lyrics-plus");
```

**Review by kyrie25 — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	const button = new Spicetify.Playbar.Button(
+		"Lyrics Plus",
+		`<svg role="img" height="16" width="16" aria-hidden="true" viewBox="0 0 16 16" data-encore-id="icon" class="Svg-sc-ytk21e-0 Svg-img-16-icon"><path d="M13.426 2.574a2.831 2.831 0 0 0-4.797 1.55l3.247 3.247a2.831 2.831 0 0 0 1.55-4.797zM10.5 8.118l-2.619-2.62A63303.13 63303.13 0 0 0 4.74 9.075L2.065 12.12a1.287 1.287 0 0 0 1.816 1.816l3.06-2.688 3.56-3.129zM7.12 4.094a4.331 4.331 0 1 1 4.786 4.786l-3.974 3.493-3.06 2.689a2.787 2.787 0 0 1-3.933-3.933l2.676-3.045 3.505-3.99z"></path></svg>`,
+		() => {
+			Spicetify.Platform.History.location.pathname !== "/lyrics-plus"
+				? Spicetify.Platform.History.push("/lyrics-plus")
+				: Spicetify.Platform.History.goBack();
+		},
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:14]:
```suggestion
		() => Spicetify.Platform.History.location.pathname !== "/lyrics-plus" ? Spicetify.Platform.History.push("/lyrics-plus") : Spicetify.Platform.History.goBack(),
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+	const style = document.createElement("style");
+	const css = `
+		.main-nowPlayingBar-lyricsButton { 
+			display: none !important; 
+		}
+		li[data-id="/lyrics-plus"] {
+			display: none;
+		}
+	`;
+	style.innerHTML = css;
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:29]:
```suggestion
	style.innerHTML = `
		.main-nowPlayingBar-lyricsButton { 
			display: none !important; 
		}
		li[data-id="/lyrics-plus"] {
			display: none;
		}
	`;
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+	style.classList.add("lyrics-plus:visual:playbar-button");
+
+	if (Spicetify.LocalStorage.get("lyrics-plus:visual:playbar-button") === "true") setPlaybarButton();
+	window.addEventListener("lyrics-plus", event => {
+		if (event.detail?.name === "playbar-button") {
+			event.detail.value ? setPlaybarButton() : removePlaybarButton();
+		}
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:36]:
```suggestion
		if (event.detail?.name === "playbar-button") event.detail.value ? setPlaybarButton() : removePlaybarButton();
```

File: CustomApps/lyrics-plus/PlaybarButton.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	});
+
+	if (Spicetify.Platform.History.location.pathname === "/lyrics-plus") button.active = true;
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/PlaybarButton.js:39]:
```suggestion
	button.active = Spicetify.Platform.History.location.pathname === "/lyrics-plus";
```

**Review by kyrie25 — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/PlaybarButton.js (hunk 11 above)
Comment (kyrie25) [reply CustomApps/lyrics-plus/PlaybarButton.js:39]: Actually, just merge this onto the constructor above.

**Review by kyrie25 — APPROVED** [review]

File: CustomApps/lyrics-plus/Settings.js [hunk 12]
```diff
@@ -431,6 +431,12 @@ function openConfig() {
 		react.createElement("h2", null, "Options"),
 		react.createElement(OptionList, {
 			items: [
+				{
+					desc: "Playbar button",
+					key: "playbar-button",
+					info: "Replace Spotify's lyrics button for one that opens lyrics-plus.",
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/Settings.js:437]:
```suggestion
					info: "Replace Spotify's lyrics button Lyrics Plus'",
```

**Review by kyrie25 — CHANGES_REQUESTED** [review]
Last one, wording is kinda weird

**Review by rxri — APPROVED** [review]

**Review by kyrie25 — APPROVED** [review]

**Pull request merged**